android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -786,15 +791,52 @@ public void onPageStarted(WebView webView, String url, Bitmap favicon) {
 
     @Override
     public boolean shouldOverrideUrlLoading(WebView view, String url) {
-      progressChangedFilter.setWaitingForCommandLoadUrl(true);
-      dispatchEvent(
-        view,
-        new TopShouldStartLoadWithRequestEvent(
-          view.getId(),
-          createWebViewEvent(view, url)));
-      return true;
-    }
+      final RNCWebView rncWebView = (RNCWebView) view;
+      final JavaScriptContextHolder jsContext = ((ReactContext) view.getContext()).getJavaScriptContextHolder();
 
+      if (jsContext.get() != 0 && rncWebView.mCatalystInstance != null) {
+        final Pair<Integer, AtomicReference<RNCWebViewModule.ShouldOverrideUrlLoadingLock.ShouldOverrideCallbackState>> lock = RNCWebViewModule.shouldOverrideUrlLoadingLock.getNewLock();
```

Comment:
I'd break out all this logic inside this if statement to another method so that the overall structure of shouldOverrideUrlLoading is very easy to understand.

Reply:
I don't agree that this would help much (not a fan of extraction into a function just cause... especially if it's not reusable), maybe let's see where the other changes take it first.